PerfGraph Live Print

friedmud opened the pull request:
Supersedes #15476

This does several things:

1. Removes the need to register 99% of perf sections (a couple of corner cases still need it)
2. Allows for recursive timed printing
3. All timing sections are now timed prints

To use this... you no longer need to register a section.  All you do is the following:

```
TIME_SECTION("shortName", 4, "Doing Short Name"
```

The `shortName` will be shown in PerfGraph print tables.  The `4` is the "level" (just like before).  `Doing Short Name` is a message that will show if the time of the section, or the memory created within the section, is too high (similar to the name you would have used for `TimedPrint` before.  That's it.

I still need to update some documentation - but I'm putting this up here to start the testing rounds on it (and get feedback).

Closes #15444

friedmud committed:
Add memory increment to TimedPrint output refs #15444

friedmud committed:
Add memory logging to perfgraph refs #15444

Fix formatting refs #15444

friedmud committed:
Remove Children columns - add self and total memory to Perfgraph refs #15444

friedmud committed:
Initial idea for print thread not working out, going to try anonther direction refs #15444

Perfgraph Live printing working refs #15444

friedmud committed:
Can now register live print names and skip printing sections without it refs #15444

friedmud committed:
Move to external PerfGraphLivePrint object refs #15444

All working refs #15444

Some cleanup refs #15444

Better spacing refs #15444

friedmud committed:
Use a condition_variable to trigger quicker printing of finishing sections refs #15444

friedmud committed:
Prevent overlapping output from live printing refs #15444

WIP PerfGraphRegistry refs #15444

friedmud committed:
Switch to external PerfGraphRegistry refs #15444

friedmud committed:
Updated FEProblemBase to new TIME_SECTION refs #15444

Remove more perfgraph registration refs #15444

More perfgraph registrations moving to new method #15444

More API usage fixups and adding input file parameters to control perf_graph_live refs #15444

Everything working refs #15444

friedmud committed:
Add in all the timers Yaqi wanted closes #15444

friedmud committed:
Add the ability to set the time and memory limits for perfgraph live refs #15444

friedmud committed:
Move to everyone having their own console ostringstream refs #15444

All working refs #15444

friedmud committed:
Add a bunch of line endings and flushes refs #15444

Fixups after rebase refs #15444

Fixing up root node related things refs #15444

Satisfy precheck refs #15444

Fixup testing issues refs #15444

Fix non-printing messages refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/632724/) on e56abc6 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format db57ca97acec2b11f7314f77e71a3a0ac1716df9`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/633299/) on bf34c99 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format db57ca97acec2b11f7314f77e71a3a0ac1716df9`

GiudGiud commented on framework/doc/content/source/utils/PerfGraph.md:
````diff
@@ -99,22 +99,22 @@ Some other capability the `PerfGraph` has is the ability to print formatted tabl
 ```
 
-`Self` time is the time actually taken by the section while `Children` time is the cumulative time of all of the sub-sections below that section and `Total` is the sum of the two.  The `%` columns represent the percent of the total run-time of the app for the number in the column to the left.
+`Calls` is the number of times that section was run. `Self` time is the time actually taken by the section while `Children` time is the cumulative time of all of the sub-sections below that section and `Total` is the sum of the two.  The `Avg` and `%` columns represent the average and percent of the total run-time of the app for the number in the column to the left. `Mem` is the memory (in Megaby ...
````
Also I know it cant really be controlled because of the "Section" column but this is already 142 char long, which is on the longer side for console width

GiudGiud replied:
Ok saw the discussion in the other PR, children wont be included it seems.

GiudGiud commented on framework/src/utils/PerfGraph.C:
```diff
@@ -146,29 +217,67 @@ PerfGraph::push(const PerfID id)
 
   _stack[_current_position] = new_node;
+
+  // Add this to the exection list
```
```suggestion
  // Add this to the execution list
```

friedmud replied:
👍

GiudGiud commented on framework/src/utils/PerfGraph.C:
```diff
@@ -344,27 +475,44 @@ PerfGraph::print(const ConsoleStream & console, unsigned int level)
+      VariadicTableColumnFormat::FIXED,   // Avg.
+      VariadicTableColumnFormat::PERCENT, // %
+      VariadicTableColumnFormat::AUTO,    // Memory
+  });                                     // %
```
```suggestion
  });
```

friedmud replied:
👍

GiudGiud commented on framework/src/utils/PerfGraphLivePrint.C:
```diff
@@ -0,0 +1,308 @@
+  {
+    _console << std::string(2 * section_increment._print_stack_level, ' ') << message;
+
+    // If we're not printind dots - just finish the line
```
```suggestion
    // If we're not printing dots - just finish the line
```

friedmud replied:
👍

GiudGiud reviewed (commented):
quick review, have not run it.

Any reason for all the new flushes? Was this an issue?

friedmud committed:
Make PerfGraphRegistry be local to the app instance refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/694430/) on 7fa31af wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b9c372b9e72d6139ba70b537316c6312e6ef4e95`

friedmud committed:
Revert "Make PerfGraphRegistry be local to the app instance refs #15444"

This reverts commit 8f5c78611a8ac11cdfcddd286a55120086dd48bb.

friedmud committed:
Almost all working refs #15444

Fixup a couple of tests refs #15444

Remove debugging statement refs #15444

Clang format #15444

Fixup build error refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/695242/) on e1a8c59 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b9c372b9e72d6139ba70b537316c6312e6ef4e95`

friedmud committed:
Remove perfgraph unit tests refs #15444

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/695380/) on aae85c2 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16298/site)

This comment will be updated on new commits.

friedmud committed:
Make sure perfgraph doesn't print during dump refs #15444

loganharbour commented on framework/include/utils/PerfGraphLivePrint.h:
```diff
@@ -0,0 +1,81 @@
+
+  const unsigned int WRAP_LENGTH = 90;
+
+  PerfGraph & _perf_graph;
+
+  std::array<PerfGraph::SectionIncrement, MAX_EXECUTION_LIST_SIZE> & _execution_list;
+
+  std::future<bool> _done_future;
+
+  const std::atomic<bool> & _destructing;
+
+  bool _should_print;
+
+  std::unordered_map<PerfID, PerfGraph::SectionInfo> & _id_to_section_info;
+
+  /// Limit (in seconds) before printing
+  Real & _time_limit;
+
+  /// Limit (in MB)
+  unsigned int & _mem_limit;
+
+  /// This is one beyond the last thing on the stack
+  unsigned int _stack_level;
+
+  /// The current stack for what the print thread has seen
+  std::array<PerfGraph::SectionIncrement *, MAX_STACK_SIZE> _print_thread_stack;
+
+  unsigned int _current_execution_list_end;
+
+  unsigned int _current_execution_list_last;
+
+  unsigned int _last_execution_list_end;
+
+  PerfGraph::SectionIncrement * _last_printed_increment;
+
+  /// The output count from the console the last time we printed
+  unsigned long long int _last_num_printed;
+
+  /// Whether or not printing happened in this iteration
+  bool _printed;
```
Docstring it up

loganharbour commented on framework/include/utils/PerfGraphRegistry.h:
```diff
@@ -0,0 +1,90 @@
+// Forward Declarations
+class PerfGraphRegistry;
+
+PerfGraphRegistry & getPerfGraphRegistry();
```
If this is gonna be global - should probably provide some info about it in a comment

loganharbour commented on framework/include/utils/PerfNode.h:
```diff
@@ -37,17 +38,33 @@ class PerfNode
+   * Get the currnet start time
+   * Only makes sense if this node is running.
+   */
+  const std::chrono::time_point<std::chrono::steady_clock> & startTime() { return _start_time; }
```
```suggestion
  const std::chrono::time_point<std::chrono::steady_clock> & startTime() const { return _start_time; }
```

loganharbour commented on framework/include/utils/PerfNode.h:
```diff
@@ -37,17 +38,33 @@ class PerfNode
+   * Get the current start memory
+   * Only makes sense if this node is running.
+   */
+  const long unsigned int & startMemory() { return _start_memory; }
```
```suggestion
  const long unsigned int & startMemory() const { return _start_memory; }
```

loganharbour commented on framework/include/utils/TimedPrint.h:
```diff
@@ -81,8 +82,8 @@ class TimedPrint final
 #endif
-    if (_active_instance)
-      return;
+    // if (_active_instance)
+    return;
```
What's going on?

friedmud replied:
Well - I was just deactivating the TimedPrints... I think I will just remove the object altogether now though.

loganharbour commented on framework/src/actions/CommonOutputAction.C:
```diff
@@ -224,6 +230,14 @@ CommonOutputAction::act()
          _app.getParam<bool>("timing")))
       create("PerfGraphOutput");
 
+    if (!_app.getParam<bool>("no_timing") && getParam<bool>("perf_graph_live"))
+      _perf_graph.setLivePrintActive(true);
+    else
+      _perf_graph.setLivePrintActive(false);
+
+    _perf_graph.setLiveTimeLimit(getParam<Real>("perf_graph_live_time_limit"));
+    _perf_graph.setLiveMemoryLimit(getParam<unsigned int>("perf_graph_live_mem_limit"));
```
```suggestion
    _perf_graph.setLivePrintActive(!_app.getParam<bool>("no_timing") && getParam<bool>("perf_graph_live"));
    _perf_graph.setLiveTimeLimit(getParam<Real>("perf_graph_live_time_limit"));
    _perf_graph.setLiveMemoryLimit(getParam<unsigned int>("perf_graph_live_mem_limit"));
```

friedmud replied:
It's a coin toss on this - but I think I would rather see the logic more clearly written out.

loganharbour commented on framework/src/actions/MeshOnlyAction.C:
```diff
@@ -36,7 +36,7 @@ MeshOnlyAction::act()
 
   // Print information about the mesh
-  _console << mesh_ptr->getMesh().get_info(/* verbosity = */ 2);
+  _console << mesh_ptr->getMesh().get_info(/* verbosity = */ 2) << std::endl;
```
I think this has an extra line already - flush instead?

friedmud replied:
It is needed.  here is what it looks like now:

![image](https://user-images.githubusercontent.com/870705/123451100-a2539a80-d59a-11eb-96d4-5240609ce943.png)

loganharbour commented on framework/src/actions/SetupMeshAction.C:
```diff
@@ -230,6 +230,8 @@ SetupMeshAction::act()
   // Create the mesh object and tell it to build itself
   if (_current_task == "setup_mesh")
   {
+    TIME_SECTION("SetupMeshAction::act::setup_mesh", 1, "Setting Up Mesh", true);
```
Why the switch to all caps? I thought before that we had only the first letter capitalized

friedmud replied:
It looks much better when they are all caps.   It's consistent through the whole framework.

loganharbour commented on framework/src/base/MooseApp.C:
```diff
@@ -710,14 +709,26 @@ MooseApp::setupOptions()
+               << formatter.toString(tree.getRoot()) << "\n### END DUMP DATA ###" << std::endl;
     _ready_to_exit = true;
+
+    _perf_graph.setLivePrintActive(true);
```
For all of these false -> true, I believe you're missing the user set state. If the user sets to false in the output action, you'll still end up with true here.

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -24,58 +26,72 @@
   updateTiming();
 
-  auto section_it = _section_time.find(section_name);
+  std::string real_section_name = section_name;
+
+  if (section_name == "Root")
+    real_section_name = _root_name;
+
+  auto section_it = _section_time.find(real_section_name);
```
```suggestion
  auto section_it = _section_time.find(section_name == "Root" ? _root_name : section_name);
```

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -97,7 +113,12 @@ PerfGraph::getTime(const TimeType type, const std::string & section_name)
   updateTiming();
 
-  auto section_it = _section_time.find(section_name);
+  std::string real_section_name = section_name;
+
+  if (section_name == "Root")
+    real_section_name = _root_name;
+
+  auto section_it = _section_time.find(real_section_name);
```
```suggestion
  auto section_it = _section_time.find(section_name == "Root" ? _root_name : section_name);
```

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -133,21 +154,71 @@ PerfGraph::getTime(const TimeType type, const std::string & section_name)
-  auto new_node = _stack[_current_position]->getChild(id);
+  PerfNode * new_node = nullptr;
+
+  if (_current_position == -1) // Must be the root node - need to makew it
```
```suggestion
  if (_current_position == -1) // Must be the root node - need to make it
```

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -158,29 +229,67 @@ PerfGraph::push(const PerfID id)
   _stack[_current_position] = new_node;
+
+  // Add this to the exection list
+  if ((_live_print_active || _live_print_all) && _pid == 0 &&
```
Doesn't `_live_print_all` here circumvent where you purposely disabled  it because of the clobbering?

friedmud replied:
Yes - `_live_print_all` should trump everything.

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -253,32 +374,37 @@ PerfGraph::recursivelyPrintGraph(PerfNode * current_node,
-    auto children = std::chrono::duration<double>(current_node->childrenTime()).count();
-    auto children_avg = children / static_cast<Real>(num_calls);
-    auto children_percent = 100. * children / total_root_time;
+    //    auto children = std::chrono::duration<double>(current_node->childrenTime()).count();
+    //    auto children_avg = children / static_cast<Real>(num_calls);
+    //    auto children_percent = 100. * children / total_root_time;
```
?

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -237,13 +357,14 @@ PerfGraph::recursivelyPrintGraph(PerfNode * current_node,
+  mooseAssert(_id_to_section_info.find(current_node->id()) != _id_to_section_info.end(),
               "Unable to find section name!");
 
-  auto & name = current_node->id() == 0 ? _root_name : _id_to_section_name[current_node->id()];
+  auto & name = _id_to_section_info[current_node->id()]._name;
 
-  mooseAssert(_id_to_level.find(current_node->id()) != _id_to_level.end(), "Unable to find level!");
-  auto & node_level = _id_to_level[current_node->id()];
+  mooseAssert(_id_to_section_info.find(current_node->id()) != _id_to_section_info.end(),
+              "Unable to find level!");
+  auto & node_level = _id_to_section_info[current_node->id()]._level;
```
This can all be replaced with:

```
mooseAssert(_id_to_section_info.count(current_node->id()), "Could not find");
const auto & entry = _id_to_section_info[current_node->id()];
auto & name = entry._name;
auto & node_level = entry._level;
```

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -253,32 +374,37 @@ PerfGraph::recursivelyPrintGraph(PerfNode * current_node,
-                  children_avg,
-                  children_percent,
+                  self_memory,
+                  //                  children,
+                  //                  children_avg,
+                  //                  children_percent,
```
?

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -295,37 +421,42 @@ PerfGraph::recursivelyPrintHeaviestGraph(PerfNode * current_node,
               "updateTiming() must be run before recursivelyPrintGraph!");
 
-  auto & name = current_node->id() == 0 ? _root_name : _id_to_section_name[current_node->id()];
+  auto & name = _id_to_section_info[current_node->id()]._name;
```
```suggestion
  mooseAssert(_id_to_section_info.count(current_node->id()), "Could not find");
  auto & name = _id_to_section_info[current_node->id()]._name;
```

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -295,37 +421,42 @@ PerfGraph::recursivelyPrintHeaviestGraph(PerfNode * current_node,
-  auto children = std::chrono::duration<double>(current_node->childrenTime()).count();
-  auto children_avg = children / static_cast<Real>(num_calls);
-  auto children_percent = 100. * children / total_root_time;
+  //  auto children = std::chrono::duration<double>(current_node->childrenTime()).count();
+  //  auto children_avg = children / static_cast<Real>(num_calls);
+  //  auto children_percent = 100. * children / total_root_time;
```
?

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -295,37 +421,42 @@ PerfGraph::recursivelyPrintHeaviestGraph(PerfNode * current_node,
-                children_avg,
-                children_percent,
+                self_memory,
+                //                children,
+                //                children_avg,
+                //                children_percent,
```
?

loganharbour commented on framework/src/utils/PerfGraph.C:
```diff
@@ -356,27 +487,44 @@ PerfGraph::print(const ConsoleStream & console, unsigned int level)
-                    "Avg(s)",
-                    "%",
+                    "Mem(MB)",
+                    // "Children(s)",
+                    // "Avg(s)",
+                    // "%",
```
?

loganharbour commented on framework/src/utils/PerfGraphLivePrint.C:
```diff
@@ -0,0 +1,309 @@
+         _last_printed_increment->_state == PerfGraph::IncrementState::printed &&
+         _id_to_section_info[_last_printed_increment->_id]._print_dots))
+      _console << '\n';
+
+    _console << std::string(2 * section_increment_start._print_stack_level, ' ') << "Finished "
+             << message;
+
+    num_horizontal_chars += std::string("Finished ").size();
```
```suggestion
    const std::string finished("Finished ");
    _console << std::string(2 * section_increment_start._print_stack_level, ' ') << finished
             << message;

    num_horizontal_chars += finished.size();
```

loganharbour commented on framework/src/utils/PerfGraphLivePrint.C:
```diff
@@ -0,0 +1,309 @@
+void
+PerfGraphLivePrint::printLiveMessage(PerfGraph::SectionIncrement & section_increment)
+{
+  auto message = !_id_to_section_info[section_increment._id]._live_message.empty()
```
```suggestion
  const auto message = !_id_to_section_info[section_increment._id]._live_message.empty()
```

loganharbour commented on framework/src/utils/PerfGraphLivePrint.C:
```diff
@@ -0,0 +1,309 @@
+
+void
+PerfGraphLivePrint::printLiveMessage(PerfGraph::SectionIncrement & section_increment)
+{
```
Should add something like:

```
mooseAssert(_id_to_section_info.count(section_increment.id), "Not found in map");
const auto & info = _id_to_section_info[section_increment.id];
```

and use it in the sections that follow. Same with in the `printStats` method

friedmud committed:
Initialize vars refs #15444

Address comments refs #15444

Another initialization refs #15444

friedmud commented on framework/include/base/MooseApp.h:
```diff
@@ -900,6 +900,9 @@ class MooseApp : public ConsoleStreamInterface, public libMesh::ParallelObject
   /// OutputWarehouse object for this App
   OutputWarehouse _output_warehouse;
 
+  /// The PerfGraph object for this applciation
```
👍

friedmud commented on framework/include/outputs/ConsoleStream.h:
```diff
@@ -80,18 +82,30 @@ class ConsoleStream
 
   /// The stream for buffering the message
-  std::ostringstream & _oss;
+  /// This stupidly has to be a shared poitner because
```
👍

friedmud commented on framework/src/systems/NonlinearSystem.C:
```diff
@@ -199,6 +199,8 @@ NonlinearSystem::solve()
   solver.set_snesmf_reuse_base(_fe_problem.useSNESMFReuseBase());
 #endif
 
+  //  _console << "\n\n" << std::endl;
```
👍

friedmud commented on framework/src/utils/PerfGraph.C:
```diff
@@ -146,29 +217,67 @@ PerfGraph::push(const PerfID id)
 
   _current_position--;
+
+  // Add this to the exection list
```
👍 ``

friedmud commented on framework/src/outputs/OutputWarehouse.C:
```diff
@@ -152,8 +156,8 @@ OutputWarehouse::outputStep(ExecFlagType type)
-   * PetscOutput::petscNonlinearOutput()
-   * PetscOutput::petscLinearOutput()
+   * Petsc Output::petscNonlinearOutput()
+   * Petsc Output::petscLinearOutput()
```
👍

friedmud commented on framework/src/utils/PerfGraph.C:
```diff
@@ -146,29 +217,67 @@ PerfGraph::push(const PerfID id)
+    // Tell the printing thread that a section has finished
+    //
+    // Note: no mutex is needed here because we're using an atomic
+    // in the predacate of the condition_variable in the thread
```
👍

friedmud commented on framework/include/utils/PerfGraph.h:
```diff
@@ -149,6 +177,39 @@ class PerfGraph
     return _section_time[section_name]._total;
   }
 
+  /**
+   * Get a reference to the self memory usage for a section
+   *
+   * This reference can be held onto and the value
+   * will be updated anyting updateTiming() is called
+   */
+  const long int & getSelfMemory(const std::string & section_name)
+  {
+    return _section_time[section_name]._self_memory;
+  }
+
+  /**
+   * Get a reference to the children memory usage for a section
+   *
+   * This reference can be held onto and the value
+   * will be updated anyting updateTiming() is called
+   */
+  const long int & getChildrenMemory(const std::string & section_name)
+  {
+    return _section_time[section_name]._children_memory;
+  }
+
+  /**
+   * Get a reference to the total memory usage for a section
+   *
+   * This reference can be held onto and the value
+   * will be updated anyting updateTiming() is called
+   */
+  const long int & getTotalMemory(const std::string & section_name)
+  {
+    return _section_time[section_name]._total_memory;
+  }
+
```
These may (probably will be) grabbed before the sections are run... so they have to be this way.

loganharbour replied:
Ah yeah, got it. The values should be default constructed to zero, correct? If so - having a zero value if it doesn't exist seems reasonable enough.

friedmud replied:
Yep - they will default to zero - should be good.

friedmud commented on framework/include/utils/PerfGraph.h:
```diff
@@ -237,29 +336,44 @@ class PerfGraph
   /// The root node of the graph
   std::unique_ptr<PerfNode> _root_node;
 
+  /// Name of the root node
+  std::string _root_name;
+
+  /// The id for the root node
+  PerfID _root_node_id;
```
?  They are used in `PerfGraph.C`

loganharbour replied:
Duh. Did ctrl + f on the changed but PerfGraph was collapsed.

friedmud committed:
Remove TimedPrint refs #15444

Clang format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/698473/) on 2214858 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b9c372b9e72d6139ba70b537316c6312e6ef4e95`

friedmud committed:
Fixup Modules refs #15444

clang format refs #15444

Splitting for debugging refs #15444

More breaking out refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/699918/) on 41774b3 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b9c372b9e72d6139ba70b537316c6312e6ef4e95`

friedmud committed:
Keep timing from happening within threads refs #15444

Undo: Testing Valgrind

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/703955/) on ba9791d wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b9c372b9e72d6139ba70b537316c6312e6ef4e95`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/704182/) on b19557e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b9c372b9e72d6139ba70b537316c6312e6ef4e95`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/704210/) on 7ed5f31 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ca37865364229b2eea17af1b310d77371c2ccf13`

friedmud committed:
Allow LivePrint to be permanently disabled refs #15444

clang format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/707588/) on a7dd89c wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 51d695e0f2b8be128832b07eacc38705bda3b6ae`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/753398/) on 8ca62ae wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a3b504b3c0fe8e1d0fcf01269004b46f52563251`

friedmud committed:
Resolve uninitialized memory issues refs #15444

Fix printing error refs #15444

Clang format refs #15444

Fix compile issue refs #15444

andrsd commented on framework/src/utils/MemoryUtils.C:
```diff
@@ -89,17 +95,32 @@ getMemoryStats(Stats & stats)
+
+    // Handle the case where we didn't get enough values by just zeroing everything
+    // since we probably got junk
+    if (i != 22)
```
`22 == val.size() + 1`?

andrsd commented on framework/src/utils/MemoryUtils.C:
```diff
@@ -143,10 +164,12 @@ getMemoryStats(Stats & stats)
 
   // page faults
-  stats._page_faults = val[index_page_faults];
+  stats._page_faults = 100; // val[index_page_faults];
```
The 3 hunks above look like leftovers from some debugging.

Was this supposed to be back to `sysconf(_SC_PAGE_SIZE)`, `val[index_virtual_size]` and `val[index_page_faults]`? If not, then delete the commented out code and will need to explain those magic numbers.

andrsd commented on framework/src/utils/PerfGraph.C:
```diff
@@ -158,29 +244,73 @@ PerfGraph::push(const PerfID id)
+      (!_id_to_section_info[id]._live_message.empty() || _live_print_all))
+  {
+    addToExecutionList(id, IncrementState::started, current_time, start_memory);
+  }
```
Do not need the  `{` and  `}`

andrsd commented on framework/src/utils/PerfGraphLivePrint.C:
```diff
@@ -0,0 +1,335 @@
+  */
+
+  // auto memory_increment = section_increment_finish._memory
+  // - section_increment_start._memory;
```
Is this just a left over from debugging?

andrsd commented on framework/src/utils/PerfGraphLivePrint.C:
```diff
@@ -0,0 +1,335 @@
+  _console << " MB";
+  _console << COLOR_DEFAULT;
+  _console << ']';
+  */
```
Leftover from testing/debugging?

andrsd reviewed (commented):
Nice cleanup :+1:

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/760735/) on 19ee52f wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format edd1406b1331568fb4ddb6ae43dbac9a1e346e0c`

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit c02af017673655633edfc4ebf8454f46bd94f059
+Subproject commit ed76100f41840c34c146dd0a7f1ff5410370b489
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/760781/) on 4a9d4f9 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format edd1406b1331568fb4ddb6ae43dbac9a1e346e0c`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/760940/) on 7623d56 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format edd1406b1331568fb4ddb6ae43dbac9a1e346e0c`

friedmud commented on framework/include/base/MooseApp.h:
```diff
@@ -918,6 +915,12 @@ class MooseApp : public ConsoleStreamInterface, public libMesh::ParallelObject
+  /// The PerfGraph object for this application
+  PerfGraph _perf_graph;
+
+  /// The RankMap is a useful object for determining how
```
Hah - that was the original comment (look at the diff above) that just got copied here.  I'll fix it.

friedmud committed:
remove mutex refs #15444

friedmud committed:
Fix module compilation refs #15444

friedmud commented on framework/include/outputs/ConsoleStream.h:
```diff
@@ -80,18 +82,30 @@ class ConsoleStream
   std::ios_base::fmtflags flags(std::ios_base::fmtflags new_flags) const;
 
+  /**
+   * The number of times something has been printed
```
Ummm - how to be specific?  This is a general printing capability - anything could be printed.

friedmud commented on framework/include/outputs/OutputWarehouse.h:
```diff
@@ -205,6 +214,9 @@ class OutputWarehouse
    */
   void solveSetup();
 
+  /// The number of times something has been printed
```
I don't get it - how am I supposed to know what is printed.  It's "something"... or "anything".  I don't understand.

friedmud committed:
Make it so that the nodetoelem map can still be timed in threads refs #15444

friedmud committed:
Remove timing from places where it shouldnt be in StochasticTools refs #15444

Update framework/src/utils/PerfGraphLivePrint.C

Use a const string for Finished

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

Update framework/src/utils/PerfGraphLivePrint.C

Change message to const

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

Update framework/src/utils/PerfGraph.C

Simplify

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

Removing more timers from stochastic tools refs #15444

Fixup some debugging stuff left around refs #15444

Initialize memory refs #15444

clang format refs #15444

Fix memory utils refs #15444

Update framework/src/utils/PerfGraph.C

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

friedmud commented:
The Griffin PR is created and ready for these changes: https://gitlab.software.inl.gov/idaholab/griffin/-/merge_requests/262

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/763923/) on 9c6bb5e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b34f0ba441b93fabfc02f48ed692869c06015399`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/763951/) on 3e1b254 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b34f0ba441b93fabfc02f48ed692869c06015399`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/763986/) on cf8a1a8 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 62de8aec37228bbd7fa92345acb2e8e88992a789`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/764078/) on cc79aaf wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 62de8aec37228bbd7fa92345acb2e8e88992a789`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/764901/) on f0c20db wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format cbc8360aa9fb66dee82a02d92dc6a5c4a6503cdb`

friedmud commented:
> quick review, have not run it.
> 
> Any reason for all the new flushes? Was this an issue?

I know this is an old comment - but we need to start talking about it.

In this new design, `_console` won't output anything until you send either `std::endl` or `std::flush`.  This is directly the opposite of the old design originally put in place by @aeslaughter .

The reason for this is that I needed to keep the streams separate from every object - so that the printing thread could print and what it printed wouldn't get jumbled up in the middle of what something else was printing.  So now: you can 100% know that if you write a bunch of stuff to `_console` without sending a `std::endl` or `std::flush` until the end... it will ALL come out together - no matter what - and never be mixed with what the PerfGraphLivePrint is doing.

Probably need to get eyeballs on this change from @idaholab/moose-team 

This will definitely need to be documented and dissiminated to our users.  Luckily - user code doesn't print very often, so it's not very disruptive.

friedmud committed:
Better error handling in memory utils refs #15444

clang-format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/766181/) on abb05c9 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format e406f586cb28126319909f20b35a4142c85b0f52`

friedmud committed:
Add some asserts refs #15444

Remove unused constant refs #15444

Fix capitilization for enum refs #15444

Resolve review comments refs #15444

Fix compile errors refs #15444

Just debugging refs #15444

Reduce duplicated code refs #15444

Add debugging refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/768161/) on 256a66e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 71ac6fd78e759ddcb10e047c2b8aefdb8b7b3cd5`

dschwen commented on modules/fluid_properties/src/userobjects/FluidPropertiesInterrogator.C:
```diff
@@ -131,7 +131,7 @@ FluidPropertiesInterrogator::execute()
       Moose::out << "**START JSON DATA**\n";
       Moose::out << json << "\n";
-      Moose::out << "**END JSON DATA**\n";
+      Moose::out << "**END JSON DATA**" << std::endl;
```
Somebody really loves `std::endl` :-). Do we really need all those flushes?

loganharbour replied:
The flushes are required to make sure that the perf output doesn't clobber AFAIK

dschwen replied:
Curious, I'd have thought output through the same stream always retains its order. I thought the rule is to flush as little as possible. Some exceptions are flushing right before user input is expected, or after a task is completed and a period of inactivity is expected (to make sure log files are up to date).

YaqiWang replied:
If the flush is not in something like an element loop, we do not need to care imo.

aeslaughter replied:
@permcody knows the reason, but I do believe we want to minimizing the flushing. All I can remember is my dad telling me not to waste water...if it is yellow let it mellow, if it is brown flush it down

roystgnr replied:
The same stream always retains its order ... with output mixed between two streams that eventually get interleaved, though, like the typical stdout+stderr pairing, less frequent flushing changes the ordering of results.  Usually you want stderr to be flushed as frequently as is sensible since that's where you put your "something's going down you need to know right now" messages, whereas with stdout you can heavily buffer most of its "work proceeding as planned, you can check it out when it's all done" messages.

roystgnr replied:
Oh, but about the interleaving - often you want to flush stdout *because of* (and immedately prior to) an error message to stderr, so that the stderr message has all of its context displayed, even if that context wasn't originally expected to be urgent when it got stuffed in the stdout buffer.

friedmud replied:
Firstly: @dschwen ... `std::endl` and `std::flush` were only added at the very end of long prints - never in tight loops.  There _should_ have been flushes there anyway... for one thing, we don't want the OS buffering stuff forever and users not getting any output.  Too many endls or flushes is certainly not the objective though: they should be used when they should be used.

@roystgnr : The stream does not always retain its order when doing _threaded_ output.  In this PR there is now a _thread_ that is also printing to std:cout... so you can literally get garbarge output.  I've used locks to make sure that actual garbage doesn't come out - but we still don't want normal blocks of text being printed to get PerfGraph stuff interleaved into it too.

To alleviate this - all objects are writing to their own stringstreams... and the contents of that stringstream are only sent to the console once a std::endl or std::flush is encountered.  In this way we can actually guarantee that an entire block of text being written at one time by an object will come out on the console as a contiguous block of text with NO PerfGraph stuff in the middle of it.

BTW: The flushes were always happening (even in the current design)... they were just being done inside ConsoleOutput.

friedmud committed:
Working on thread safety refs #15444

Fixup memory errors refs #15444

clang format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/794170/) on 30da31f wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 3ae0888adb16b710eea21a0d09ddde340bcbb123`

friedmud committed:
Fix abort messages not printing refs #15444

friedmud committed:
Documentation update refs #15444

friedmud committed:
Test PerfGraphLivePrint refs #15444

friedmud committed:
Add deprecated timed prints refs #15444

clang format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/795012/) on ae03c20 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format d4199c27c017d762bbddfd8e91b5f9eed90a7c8d`

friedmud committed:
Switch to using a vector for storing SectionInfo and allow a lockless way for PerfGraph to read from the registry refs #15444

Clang-format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/796073/) on a638ff7 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 686640043e5d78b3947e4ffa55910770a94ae985`

friedmud committed:
Add PerfGraphLivePrint doco refs #15444

friedmud committed:
Change how the print stack works to make it more robust, modify how the PGLP is told to stop to make it more efficient refs #15444

clang format refs #15444

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/797523/) on 51fa7b1 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/16298/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/16298/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 49ea6f56b00dfbda3d508fe63e91ad6cecffc60a`

loganharbour commented:
Slowly working through a review. I added the remainder of the tests on next + code coverage. This is a pretty big change, good to do it now

moosebuild commented:
Job [Code coverage](https://civet.inl.gov/job/797653/) on c738058 wanted to post the following:

Generated code coverage:<br><br>[moose](https://mooseframework.inl.gov/docs/PRs/16298/coverage/moose)<br>[chemical_reactions](https://mooseframework.inl.gov/docs/PRs/16298/coverage/chemical_reactions)<br>[combined](https://mooseframework.inl.gov/docs/PRs/16298/coverage/combined)<br>[contact](https://mooseframework.inl.gov/docs/PRs/16298/coverage/contact)<br>[external_petsc_solver](https://mooseframework.inl.gov/docs/PRs/16298/coverage/external_petsc_solver)<br>[fluid_properties](https://mooseframework.inl.gov/docs/PRs/16298/coverage/fluid_properties)<br>[fsi](https://mooseframework.inl.gov/docs/PRs/16298/coverage/fsi)<br>[functional_expansion_tools](https://mooseframework.inl.gov/docs/PRs/16298/coverage/functional_expansion_tools)<br>[geochemistry](https://mooseframework.inl.gov/docs/PRs/16298/coverage/geochemistry)<br>[heat_conduction](https://mooseframework.inl.gov/docs/PRs/16298/coverage/heat_conduction)<br>[level_set](https://mooseframework.inl.gov/docs/PRs/16298/coverage/level_set)<br>[misc](https://mooseframework.inl.gov/docs/PRs/16298/coverage/misc)<br>[navier_stokes](https://mooseframework.inl.gov/docs/PRs/16298/coverage/navier_stokes)<br>[peridynamics](https://mooseframework.inl.gov/docs/PRs/16298/coverage/peridynamics)<br>[phase_field](https://mooseframework.inl.gov/docs/PRs/16298/coverage/phase_field)<br>[porous_flow](https://mooseframework.inl.gov/docs/PRs/16298/coverage/porous_flow)<br>[ray_tracing](https://mooseframework.inl.gov/docs/PRs/16298/coverage/ray_tracing)<br>[rdg](https://mooseframework.inl.gov/docs/PRs/16298/coverage/rdg)<br>[richards](https://mooseframework.inl.gov/docs/PRs/16298/coverage/richards)<br>[stochastic_tools](https://mooseframework.inl.gov/docs/PRs/16298/coverage/stochastic_tools)<br>[tensor_mechanics](https://mooseframework.inl.gov/docs/PRs/16298/coverage/tensor_mechanics)<br>[xfem](https://mooseframework.inl.gov/docs/PRs/16298/coverage/xfem)<br>

This comment will be updated on new commits.

friedmud committed:
Fixup printing in TensorMechanics refs #15444

friedmud committed:
Fix threading error refs #15444

friedmud commented on framework/doc/content/source/utils/PerfGraph.md:
```diff
@@ -34,37 +37,81 @@ The final one is for when your object is NOT a `MooseObject` inherited object.
+
+- `section_name`: The short name of the section.  This is the name used in the final table.  It is normally the function name or some other short name.
+- `level`: The logging level
+- `live_message`: OPTIONAL - but highly recommended.  This is the message `PerfGraphLivePrint` will print to the screen (if necessary).  It should be descriptive, title-cased, and written in an active way, e.g. "Calculating Lama Heights"
```
?  I don't get it... it's spelled "Lama", right?  https://en.wikipedia.org/wiki/Lama_(genus) - it seems like both are actually things though.

friedmud committed:
Update framework/doc/content/source/utils/PerfGraphLivePrint.md

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

friedmud committed:
Update framework/include/outputs/OutputWarehouse.h

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

friedmud commented on framework/include/utils/PerfGraphLivePrint.h:
```diff
@@ -0,0 +1,121 @@
+  /// Start printing
+  void start();
+
+protected:
```
```suggestion
private:
```

friedmud committed:
Update framework/include/utils/PerfGraphLivePrint.h

friedmud commented on framework/src/problems/FEProblemBase.C:
```diff
@@ -697,32 +652,44 @@ FEProblemBase::initialSetup()
-                               std::vector<RealGradient>(getMaxQps(), RealGradient(0.)));
-    _second_phi_zero[tid].resize(_nl->getMaxVarNDofsPerElem(),
-                                 std::vector<RealTensor>(getMaxQps(), RealTensor(0.)));
+    TIME_SECTION("resizingVarValues", 5, "Resizing Variable Vlues");
```
```suggestion
    TIME_SECTION("resizingVarValues", 5, "Resizing Variable Values");
```

friedmud committed:
Update framework/src/problems/FEProblemBase.C
